### DISCUSSION TITLE
zfs send -w produces different binary dumps

### DISCUSSION THREAD

**Discussion by andaag** [openzfs/zfs#11431] [General]
Hi

I've been working on some backup scripts that does a md5 of my backup, and realized that zfs send if using -w does not produce the same output...

```
sudo zfs send -vwP bpool/BOOT/ubuntu_fnpezh@autosnap_2020-12-22_13:33:51_monthly | md5sum
```
Running this repeatedly produces different md5sum's every time.

If I dump it to a file and zfs recv it, it seems to work just fine..

This seems... scary to me. Is this expected behavior? If I remove -w it produces identical snapshots, but I'm backing up to a source I'd rather not trust if I can avoid it.

```
zfs-0.8.3-1ubuntu12.5
zfs-kmod-0.8.3-1ubuntu12.5
```

**Comment by bghira** [comment]
use zstreamdump to get a checksum instead of that method, which is not reliable.
